src/components/MultiSelect/MultiSelectMenuButton.tsx

```diff
@@ -0,0 +1,97 @@
+import React from "react";
+import Button from "./../Button/Button";
+import Icon from "./../Icons/Icon";
+import { Box, useMultiStyleConfig } from "@chakra-ui/react";
+import { SelectedItems } from "./MultiSelectTypes";
+
+export interface MultiSelectMenuButtonProps {
+  /** The id of the multiSelect using this button. */
+  multiSelectId: string;
+  /** The label of the multiSelect using this button. */
+  label: string;
+  /** The open status of the multiselect menu. */
+  isOpen: boolean;
+  /** The selected items state (items that were checked by user). */
+  selectedItems: SelectedItems;
+  /** The callback function for the menu toggle. */
+  onMenuToggle?: () => void;
+  /** The action to perform for clear/reset button of multiselect. */
+  onClear?: () => void;
+}
+
+const MultiSelectMenuButton = React.forwardRef<
+  HTMLButtonElement,
+  MultiSelectMenuButtonProps
+>((props, ref?) => {
+  const { multiSelectId, label, isOpen, onMenuToggle, selectedItems, onClear } =
+    props;
+  const styles = useMultiStyleConfig("MultiSelect", {});
+  const iconType = isOpen ? "minus" : "plus";
+
+  // Sets the selected items count on the menu button.
+  function getSelectedItemsCount(multiSelectId: string) {
+    if (selectedItems[multiSelectId]?.items.length > 0) {
+      return `${selectedItems[multiSelectId].items.length}`;
+    }
+    return null;
+  }
```

Comment:
Generally, this is okay. The problem is that this function is called twice on lines 63 and 73 and it'd be better if it was called once since, for both calls, the returned value is the same. 

So we can do something like
```
const countValue = getSelectedItemsCount(multiSelectId);
```
and then 
```
 {countValue && (
        <Box
          as="span"
          __css={styles.selectedItemsCountButton}
          onClick={onClear}
          onKeyPress={onKeyPress}
          role="button"
          tabIndex={0}
        >
          <Box as="span" verticalAlign="text-bottom">
            {countValue}
          </Box>
```

Another possibility is to not make it a function at all.
```
// Sets the selected items count on the menu button.
  let getSelectedItemsCount = "";
  if (selectedItems[multiSelectId]?.items.length > 0) {
    getSelectedItemsCount = `${selectedItems[multiSelectId].items.length}`;
  }
```
and then 
```
{getSelectedItemsCount && (
        <Box
          as="span"
          __css={styles.selectedItemsCountButton}
          onClick={onClear}
          onKeyPress={onKeyPress}
          role="button"
          tabIndex={0}
        >
          <Box as="span" verticalAlign="text-bottom">
            {getSelectedItemsCount}
          </Box>
```


Either approach is fine but the point is to try to call a function once if it returns the same value for the same input, and assign that value to a variable.

Reply:
fixed

Reply:
fixed